src/app/server/JointFabricDatastore.cpp

```diff
@@ -122,16 +122,166 @@ CHIP_ERROR JointFabricDatastore::RemoveNode(NodeId nodeId)
+                continue;
+            }
+            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitFailed)
+            {
+                CHIP_ERROR failureCode(it->statusEntry.failureCode);
+
+                if (failureCode == CHIP_ERROR_CONSTRAINT || failureCode == CHIP_ERROR_RESOURCE_EXHAUSTED)
+                {
+                    // remove entry from the list
+                    it = mEndpointGroupIDEntries.erase(it);
+                    continue;
+                }
+                else
+                {
+                    // If the node is pending, we need to update the status entry to Committed
+                    it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+                }
+            }
+        }
+
+        ++it;
+    }
+
+    for (auto it = mEndpointBindingEntries.begin(); it != mEndpointBindingEntries.end();)
+    {
+        if (it->nodeID == nodeId)
+        {
+            if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kPending)
+            {
+                // If the node is pending, we need to update the status entry to Committed
+                it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+            }
+            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kDeletePending)
+            {
+                // If the node is delete pending, we need to remove the entry
+                it = mEndpointBindingEntries.erase(it);
+                continue;
+            }
+            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitFailed)
+            {
+                CHIP_ERROR failureCode(it->statusEntry.failureCode);
+
+                if (failureCode == CHIP_ERROR_CONSTRAINT || failureCode == CHIP_ERROR_RESOURCE_EXHAUSTED)
+                {
+                    // remove entry from the list
+                    it = mEndpointBindingEntries.erase(it);
+                    continue;
+                }
+                else
+                {
+                    // If the node is pending, we need to update the status entry to Committed
+                    it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+                }
+            }
+        }
+
+        ++it;
+    }
+
+    // 4.
+    for (auto it = mGroupKeySetList.begin(); it != mGroupKeySetList.end();)
+    {
+        // TODO: Unclear on what to do here. Spec needs fixing
+        ++it;
+    }
 
-    // 4. TODO
+    // 5.
+    for (auto it = mACLEntries.begin(); it != mACLEntries.end();)
+    {
+        if (it->nodeID == nodeId)
+        {
+            if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kPending)
+            {
+                // If the node is pending, we need to update the status entry to Committed
```

Comment:
```suggestion
                // TODO: check if change has been made, retry if not. Mark as committed upon success.
                // If the node is pending, we need to update the status entry to Committed
```

Reply:
Added a TODO to handle this in a future PR.